webpack/scenes/Content/Table/ContentTable.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { TableVariant, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import PropTypes from 'prop-types';
+import TableWrapper from '../../../components/Table/TableWrapper';
+import { getContent } from '../ContentActions';
+import { selectContent, selectContentStatus, selectContentError } from '../ContentSelectors';
+import SelectableDropdown from '../../../components/SelectableDropdown';
+import contentConfig from '../ContentConfig';
+
+const ContentTable = ({ selectedContentType, setSelectedContentType, contentTypes }) => {
+  const status = useSelector(selectContentStatus);
+  const error = useSelector(selectContentError);
+  const response = useSelector(selectContent);
+  const [searchQuery, updateSearchQuery] = useState('');
+  const { results, ...metadata } = response;
+
+  const { columnHeaders } = contentConfig().find(type =>
+    type.names.singular === contentTypes[selectedContentType][0]);
+
+  /* eslint-disable react/no-array-index-key */
+  function buildRows(details) {
+    const rows = [];
+    if (details) {
+      columnHeaders.forEach((header, index) =>
+        rows.push(<Td key={index}>{header.getProperty(details)}</Td>));
+    }
+    return rows;
+  }
+
+  return (
+    <TableWrapper
+      {...{
+        metadata,
+        searchQuery,
+        updateSearchQuery,
+        error,
+        status,
+      }}
+      variant={TableVariant.compact}
```

Comment:
```suggestion
      key={contentTypes[selectedContentType][1]}
      variant={TableVariant.compact}
```

So I noticed the autosearch params are not updated when the selectedType changes. This will fix that.

Reply:
Changing the key will cause the entire TableWrapper to unmount and remount.  It would work but seems a bit drastic.  I'm confused because I would think adding `selectedContentType` to `additionalListeners` should be all you need?

Reply:
So the search component inside tableWrapper is not refreshed with the additionalListener..That causes the autosearch Endpoint to not change. When switching between different content types we need to update the search bar with correct endpoints..Hence the complete refresh of the TableWrapper.

Reply:
We should actually remove the selectedContentType from additionalListeners since it's not needed anymore given we are re-rendering the whole table instead.